src/main/java/io/stargate/sgv2/jsonapi/service/shredding/collections/WritableShreddedDocument.java

```diff
@@ -338,7 +343,12 @@ public void shredVectorize(JsonPath path) {
      * <p>Method will add path to {@link #existKeys}.
      */
     private void addKey(JsonPath key) {
-      existKeys.add(key);
```

Comment:
We basically use and extend existing functionality to avoid traversing multiple times, or storing duplicate Set of paths for common case: we only construct Set if there are duplicates, and then collect all duplicate paths for caller to access.